web/modules/custom/custom_weather/src/Service/UserCityHandler.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace Drupal\custom_weather\Service;
+
+use Drupal\Core\Database\Database;
+
+/**
+ * Determines the city for displaying weather.
+ */
+class UserCityHandler {
+
+  /**
+   * Returns the city from the database or sets a default value.
+   */
+  public function getCityFromDatabase() {
```

Comment:
fromDatabase in the name is unnecessary
One of the reasons for services to exist is to be able easily change parts of your application logic
And if you were to store this data anywhere but not database, name loses sense